docs/how-to/troubleshoot_deployment_issues.md

```diff
@@ -0,0 +1,199 @@
+│ 
+│   with juju_model.private5g,
+│   on main.tf line 1, in resource "juju_model" "private5g":
+│    1: resource "juju_model" "private5g" {
+│ 
+│ Unable to create model, got error: failed to open kubernetes client: annotations map[controller.juju.is/id:ced7016b-3a63-4133-8988-cf33068c3cdf
+│ model.juju.is/id:2abfe7ab-9e40-4e0d-8158-53450c47b2db] for namespace "private5g" not valid must include map[controller.juju.is/id:ced7016b-3a63-4133-8988-cf33068c3cdf
+│ model.juju.is/id:9755936b-8084-4397-8a67-28773b361dfa] (not valid)
+```
+
+### Recommended Actions
+
+Check all the existing Juju models:
+
+```shell
+$ juju models
+controller: microk8s-localhost
+
+Model       Cloud/Region        Type        Status      Units  Access  Last connection
+controller  microk8s/localhost  kubernetes  available   1       admin  just now
+private5g*  microk8s/localhost  kubernetes  destroying  -       admin  6 minutes ago
+private5g2  microk8s/localhost  kubernetes  destroying  -       admin  1 minute ago
+sdcore      microk8s/localhost  kubernetes  available   19      admin  2024-10-04
+```
+
+According to [Customize the model name using variables][Deploy SD-Core K8s with Terraform], provide a model name which does not exist in Juju controller:
+
+Create a `terraform.tfvars` file to specify the name of the Juju model to deploy to.
+
+```shell
+cat << EOF | tee terraform.tfvars
+model_name = "put your model-name here"
+EOF
+```
+
+Deploy the applications by customizing the model name attribute:
+
+```shell
+terraform apply -var-file="terraform.tfvars" -auto-approve 
+```
+
+## 4. Juju failed to deploy Charmed Aether SD-Core as Juju controller is not reachable
+
+### Symptoms
+
+While Juju is performing the deployment, the Juju controller may be unavailable which fails the deployment with the error `timeout to Juju Terraform provider` as following.
+
+```console
+$ terraform apply --auto-approve
+╷
+│ Error: Invalid provider configuration
+│ 
+│ Provider "registry.terraform.io/juju/juju" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in
+│ the provider documentation.
+│ 
+│ Error: dial tcp 10.152.183.251:17070: i/o timeout
+│ 
+│   with provider["registry.terraform.io/juju/juju"],
+│   on <empty> line 0:
+│   (source code not available)
+│ 
+│ Connection error, please check the controller_addresses property set on the provider
+```
+
+### Recommended Actions
+
+Make sure that the Juju controller is available:
+
+```shell
+$ juju controllers
+Use --refresh option with this command to see the latest information.
+
+Controller           Model      User   Access     Cloud/Region        Models  Nodes  HA  Version
+microk8s-localhost*  private5g  admin  superuser  microk8s/localhost       9      -   -  3.4.5  
+```
+
+If it does not output the controller details, please follow the guide [manage Juju controllers][Manage Juju Controller] to create a Juju controller.
```

Comment:
Wouldn't it be better to point to our documentation instead. 
https://canonical-charmed-aether-sd-core.readthedocs-hosted.com/en/latest/tutorials/getting_started/#bootstrap-a-juju-controller

Reply:
I provided the Juju page which provides more actions for controller management.  In our guide, we only show how to create a controller. There could be some cases that `juju bootstrap microk8s` may fail. I sometimes meet this issue. The reason is  `juju bootstrap microk8s` creates a controller with a default name. That controller may exists but unreachable. If we run the same command again, bootstrap fails. Our guide does not provide a solution for it but Juju page provides the solution and I do not want to repeat them here.  I prefer to change my sentence to `create an accessible Juju controller` instead of changing the reference link.